### PR TITLE
[ML] Send autodetect config updates as JSON

### PR DESCRIPTION
**Pull request by edsavage**
Send configuration updates to model plot, scheduled events, detector
rules and filters as JSON formatted strings.

Remove any now unneeded supporting code.

Note that a handful of test cases have been muted. This is only temporary and the tests will be re-enabled when the supporting C++ code is merged and available.

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/writer/AutodetectControlMsgWriter.java [hunk 1]
```diff
@@ -204,37 +224,44 @@ public void writeCategorizationStopOnWarnMessage(boolean isStopOnWarn) throws IO
...
+            stringBuilder.append(msg);
+        }
         writeMessage(stringBuilder.toString());
     }
 
     public void writeUpdateScheduledEventsMessage(List<ScheduledEvent> events, TimeValue bucketSpan) throws IOException {
         StringBuilder stringBuilder = new StringBuilder();
-        stringBuilder.append(UPDATE_MESSAGE_CODE).append("[scheduledEvents]\n");
-        if (events.isEmpty()) {
-            stringBuilder.append("clear = true\n");
```
Comment (dimitris-athanasiou) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/writer/AutodetectControlMsgWriter.java:235]: So if we now have an empty list of events, c++ will receive an empty `events` array. Will this clear any existing events?

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM Just a double-checking question :-)

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/writer/AutodetectControlMsgWriter.java (hunk 1 above)
Comment (edsavage) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/writer/AutodetectControlMsgWriter.java:235]: Yes, the C++ code has been adjusted such that an empty events array, generating the message `u{"events":[]}` will clear out any existing events.

**Comment by edsavage** [comment]
jenkins retest this please

**Pull request merged**